proplot/colors.py

```diff
@@ -2758,23 +2874,31 @@ def _init_cmap_database():
     """
     # WARNING: Skip over the matplotlib native duplicate entries
     # with suffixes '_r' and '_shifted'.
-    attr = '_cmap_registry' if hasattr(mcm, '_cmap_registry') else 'cmap_d'
-    database = getattr(mcm, attr)
-    if mcm.get_cmap is not _get_cmap:
-        mcm.get_cmap = _get_cmap
-    if mcm.register_cmap is not _register_cmap:
-        mcm.register_cmap = _register_cmap
-    if not isinstance(database, ColormapDatabase):
-        database = {
-            key: value for key, value in database.items()
-            if key[-2:] != '_r' and key[-8:] != '_shifted'
-        }
-        database = ColormapDatabase(database)
-        setattr(mcm, attr, database)
+    if hasattr(mcm, "cmap_d") or hasattr(mcm, "_cmap_registry"):
+        attr = "_cmap_registry" if hasattr(mcm, "_cmap_registry") else "cmap_d"
+        database = getattr(mcm, attr)
+        if mcm.get_cmap is not _get_cmap:
+            mcm.get_cmap = _get_cmap
+        if mcm.register_cmap is not _register_cmap:
+            mcm.register_cmap = _register_cmap
+        if not isinstance(database, ColormapDatabase):
+            database = {
+                key: value
+                for key, value in database.items()
+                if key[-2:] != "_r" and key[-8:] != "_shifted"
+            }
+            database = ColormapDatabase(database)
+            setattr(mcm, attr, database)
+    elif hasattr("mcm", "_gen_cmap_registry"):
```

Comment:
I was getting the `raise ValueError("ColormapDatabase not initialized")` error, but this needs to be 

```suggestion
    elif hasattr(mcm, "_gen_cmap_registry"):
```